be/src/vec/data_types/serde/data_type_object_serde.h

```diff
@@ -73,10 +73,7 @@ class DataTypeObjectSerDe : public DataTypeSerDe {
 
     void write_column_to_arrow(const IColumn& column, const NullMap* null_map,
                                arrow::ArrayBuilder* array_builder, int start, int end,
-                               const cctz::time_zone& ctz) const override {
-        throw doris::Exception(ErrorCode::NOT_IMPLEMENTED_ERROR,
-                               "write_column_to_arrow with type " + column.get_name());
-    }
+                               const cctz::time_zone& ctz) const override;
     void read_column_from_arrow(IColumn& column, const arrow::Array* arrow_array, int start,
```

Comment:
dose we need to support read from arrow? for some spark situation : make string value to variant data type value into doris

Reply:
currently this pr is to solve the connector problem for variant, for read from arrow we could use cast expr like cast(xxx as variant)